cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -137,13 +137,34 @@ func (c *Controller) syncSpc(spc *apis.StoragePoolClaim) error {
 
 	spcObj, err := c.populateVersion(gotSPC)
 	if err != nil {
-		klog.Errorf("failed to add versionDetails to SPC %s:%s", spc.Name, err.Error())
-		return err
+		klog.Errorf("failed to add versionDetails to spc %s:%s", gotSPC.Name, err.Error())
+		c.recorder.Event(
+			gotSPC,
+			corev1.EventTypeWarning,
+			"FailedPopulate",
+			fmt.Sprintf("Failed to add current version: %s", err.Error()),
+		)
+		return nil
 	}
+	gotSPC = spcObj
 	spcObj, err = c.reconcileVersion(spcObj)
 	if err != nil {
-		klog.Errorf("failed to upgrade SPC %s:%s", spc.Name, err.Error())
-		return err
+		klog.Errorf("failed to upgrade spc %s:%s", gotSPC.Name, err.Error())
+		c.recorder.Event(
+			gotSPC,
+			corev1.EventTypeWarning,
+			"FailedUpgrade",
+			fmt.Sprintf("Failed to upgrade spc to %s version: %s",
+				gotSPC.VersionDetails.Desired,
+				err.Error(),
+			),
+		)
+		gotSPC.VersionDetails.Status.Message = "Failed to reconcile spc version"
+		gotSPC.VersionDetails.Status.Reason = err.Error()
+		gotSPC.VersionDetails.Status.LastUpdateTime = metav1.Now()
+		_, err = c.clientset.OpenebsV1alpha1().StoragePoolClaims().Update(gotSPC)
+		klog.Errorf("failed to update versionDetails status for spc %s:%s", gotSPC.Name, err.Error())
+		return nil
 	}
 	gotSPC = spcObj
 	// assinging the latest spc object
```

Comment:
extra variable can be removed, refactored

Reply:
Done